presto-main/src/test/java/io/prestosql/sql/planner/TestUnwrapCastInComparison.java

```diff
@@ -32,830 +32,364 @@
     public void testEquals()
     {
         // representable
-        assertPlan(
-                "SELECT * FROM (VALUES SMALLINT '0') t(a) WHERE a = DOUBLE '1'",
-                output(
-                        filter("A = SMALLINT '1'",
-                                values("A"))));
+        testUnwrap("smallint", "a = DOUBLE '1'", "a = SMALLINT '1'");
 
-        assertPlan(
-                "SELECT * FROM (VALUES BIGINT '0') t(a) WHERE a = DOUBLE '1'",
-                output(
-                        filter("A = BIGINT '1'",
-                                values("A"))));
+        testUnwrap("bigint", "a = DOUBLE '1'", "a = BIGINT '1'");
 
         // non-representable
-        assertPlan(
-                "SELECT * FROM (VALUES SMALLINT '0') t(a) WHERE a = DOUBLE '1.1'",
-                output(
-                        filter("A IS NULL AND NULL",
-                                values("A"))));
-        assertPlan(
-                "SELECT * FROM (VALUES SMALLINT '0') t(a) WHERE a = DOUBLE '1.9'",
-                output(
-                        filter("A IS NULL AND NULL",
-                                values("A"))));
+        testUnwrap("smallint", "a = DOUBLE '1.1'", "A IS NULL aND NULL");
```

Comment:
is the casing of `aND` purposeful :)

Reply:
I guess you wanted to make other `A` lowercase

Reply:
how come?! it was supposed to be `\bA\b`

Reply:
fixed

Reply:
https://github.com/prestosql/presto/pull/6109